### ISSUE TITLE
Insufficient contrast ratio from dark theme

### ISSUE THREAD

**Issue by IO-Fire** [hlassiege/bloggr-demo#3]
Insufficient text contrast ratio on text and components when browser using dark theme as Mistral is light theme only.

### Example

[`ProseImg.vue`](https://github.com/bloggrify/bloggrify/blob/0bc0d709f48fadbc7e47cf8339b56418b4664baf/components/content/prose/ProseImg.vue#L10) from [@bloggrify/core](https://github.com/bloggrify/bloggrify/blob/0bc0d709f48fadbc7e47cf8339b56418b4664baf/components/content/prose/ProseImg.vue#L10) , see diff below. 

## Potential Fix

Override the prose components to remove the dark theme css classes.

`patches/@bloggrify__core.patch`
See updated version in https://github.com/bloggrify/bloggrify-mistral/issues/3#issuecomment-2525028634

**Comment by IO-Fire** [comment]
### New potential fix:

Override `app.vue` with patch 

`patches/@bloggrify__core.patch`:
> Sets the theme to light, preventing the dark theme being used by the browser.
```diff
diff --git a/app.vue b/app.vue
index 339a1b1dc243c8301739b71765794ee8260434a9..3d499934a5dac0813939de79d081cc615cce1e70 100644
--- a/app.vue
+++ b/app.vue
@@ -4,6 +4,7 @@
     </div>
 </template>
 <script setup lang="ts">
+import { useColorMode, useDark } from '@vueuse/core'
 import {useAnalytics} from '~/composables/useAnalytics'
 
 const config = useAppConfig()
@@ -47,6 +48,9 @@ useHead({
         },
     ],
 })
+
+useColorMode({initialValue: 'light'})
+useDark().value = false
 </script>
 <style lang="scss">
 @tailwind base;
```

Edit:
- Add patch description

**Comment by hlassiege** [comment]
I used lighthouse to understand the problem and apparently there is a problem of contrast in the footer.
I modified the footer to increase the level of gray and now it's 100 on lighthouse.
It should be ok now

**Comment by IO-Fire** [comment]
Thank you, @hlassiege, for your fix implementation.

Unfortunately https://github.com/bloggrify/bloggrify-mistral/commit/c3bc9f85d2c2768fa185b10eacd76156eecef8c3 only treats one of the symptoms.

Request issue be re-opened by @hlassiege.

See opening comment's example https://github.com/bloggrify/bloggrify-mistral/issues/3#issue-2722363554, [`ProseImg.vue`](https://github.com/bloggrify/bloggrify/blob/0bc0d709f48fadbc7e47cf8339b56418b4664baf/components/content/prose/ProseImg.vue#L10) from [@bloggrify/core](https://github.com/bloggrify/bloggrify/blob/0bc0d709f48fadbc7e47cf8339b56418b4664baf/components/content/prose/ProseImg.vue#L10) is inherited as a [layer](https://nuxt.com/docs/getting-started/layers). Therefore extended components and other components with dark css theming are still impacted.

### Additional Examples

- Change browser / OS to use dark mode
- Open [https://mistral.bloggrify.com/features](https://mistral.bloggrify.com/features)
- Run lighthouse report
  - See section **Accessibility** > **CONTRAST**
    - `Background and foreground colors do not have a sufficient contrast ratio.`
      - Failing Elements: 
        - `div.text-sm.text-center.text-gray-600.dark:text-gray-300.mt-1` × 2
          - Resulting component is`ProseImg`
  - [`ProseCode`](https://github.com/bloggrify/bloggrify/blob/0bc0d709f48fadbc7e47cf8339b56418b4664baf/components/content/prose/ProseCode.vue#L13)'s copy to clipboard is also impacted in page section [#choose-your-language](https://mistral.bloggrify.com/features#choose-your-language) with dark theme copy button on the normally dark code area.

Table of contents:
<img width="312" alt="image" src="https://github.com/user-attachments/assets/566ea200-3cc0-432a-a345-1a0f33497b18">

### Potential fix

Extending on from https://github.com/bloggrify/bloggrify-mistral/issues/3#issuecomment-2525028634, overriding `app.vue` has already proved to work. It can be implemented within this repo rather than as a patch.

While a potential fix for this theme may work, other themes that are missing a dark theme implementation will still be impacted. This may be a [core module](https://github.com/bloggrify/bloggrify) issue.

### Conclusion

While the potential fix sheds some *light* on the situation, it raises the point that this issue may be outwith the theme's responsibility.

## General Notes

I recommend implementing the fix in a PR rather than a direct commit. This approach allows for greater flexibility and ensures that the changes can be peer-reviewed. 😊

**Comment by hlassiege** [comment]
Ok. There is something I don't understand. I can't reproduce the issue.
I did switched my browser to dark mode (I'm on Chrome, MacOS)
![image](https://github.com/user-attachments/assets/64e68979-566c-4558-ae05-8b2576c33c55)

but the application is still in light mode and I don't have more contrast issue with and without this setting.
On your screenshot however, I see it's full dark. I suppose there is something different ?

About the fix itself, I'm not sure to understand exactly the code 

```
useColorMode({initialValue: 'light'})
useDark().value = false
```
I understand the initial value will be light.
But does this mean that the dark mode is always disabled then ? That would be a problem for themes using dark mode

**Comment by IO-Fire** [comment]
> Ok. There is something I don't understand. I can't reproduce the issue. I did switched my browser to dark mode (I'm on Chrome, MacOS) ![image](https://private-user-images.githubusercontent.com/666123/393586559-64e68979-566c-4558-ae05-8b2576c33c55.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM2NTQxODcsIm5iZiI6MTczMzY1Mzg4NywicGF0aCI6Ii82NjYxMjMvMzkzNTg2NTU5LTY0ZTY4OTc5LTU2NmMtNDU1OC1hZTA1LThiMjU3NmMzM2M1NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIwOFQxMDMxMjdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lZTU2OGRmMjE4ZWU5NDY0MmI1Y2YxOGU5MTE1MDAzZDA0NTNmOWEzOWRjMDc5ZDg1NjgwOTM3YjQ2NTNhOGNjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ca7ggGchG8eWVvyYfjgc0fk6NEdvqIc6J5E6IdKWqgY)
> but the application is still in light mode and I don't have more contrast issue with and without this setting. On your screenshot however, I see it's full dark. I suppose there is something different ?

It appears the browser's theme does not update `prefers-color-scheme`, looks like the browser uses the OS theme mode. [Emulate CSS media features](https://developer.chrome.com/docs/devtools/rendering/emulate-css#emulate_css_media_feature_prefers-color-scheme) will force regardless of OS theme but may not apply in lighthouse.

> 
> About the fix itself, I'm not sure to understand exactly the code
> 
> ```
> useColorMode({initialValue: 'light'})
> useDark().value = false
> ```

The `useColorMode` tries to prevent the initialValue being 'auto' which will switch to 'dark'. 

The `useDark().value = false` sets the the inverse of the 'dark' theme which is 'light'. This is the theme value until changed by a `useDark()` or equivalent.

> 
> I understand the initial value will be light. But does this mean that the dark mode is always disabled then ? 

##### Dark Mode Setting:
The initial `useColorMode` setting attempts to prevent the 'auto' mode which would result in 'dark', if that is the preferred mode. 

The `useDark` sets not dark mode i.e. 'light' mode.

Since Mistral has no dark theme, this behaviour is logical.

The dark mode can still be updated by `useDark()` elsewhere.

> That would be a problem for themes using dark mode

##### Patch / Override:

Patches or overrides only affect the project they originate from or extend.

For example, if Mistral patches the core module, the core project changes only when using the Mistral theme or extensions of Mistral.


**Edit**: Fix final explanations.

**Comment by IO-Fire** [comment]
> On your screenshot however, I see it's full dark. I suppose there is something different ?

@hlassiege, that is the `:toc` example in [https://mistral.bloggrify.com/features#table-of-content-with-a-shortcode](https://mistral.bloggrify.com/features#table-of-content-with-a-shortcode).

**Comment by hlassiege** [comment]
I tried the suggested fix
- on epoxia and theme using dark mode. It's ok. The dark mode is still supported
- on mistral. The problem with ProseImg is gone

I also fixed the table of content issue in dark mode 

(it's not deployed on mistral.bloggrify.com. It needs a new release first)
